website/docs/versioning.md

```diff
@@ -0,0 +1,146 @@
+
+- Copy the full `docs/` folder contents into a new `versioned_docs/version-<version>/` folder.
+- Create a versioned sidebars file based from your current [sidebar](sidebar.md) configuration (if it exists). Saved it as `versioned_sidebars/version-<version>-sidebars.json`.
+- Append the new version number into `versions.json`.
+
+## Files
+
+### Creating new files
+
+1. Place the new file into the corresponding version folder.
+1. Include the reference for the new file into the corresponding sidebar file, according to version number.
+
+**Master docs**
+
+```shell
+# The new file.
+docs/new.md
+
+# Edit the corresponding sidebar file.
+sidebar.js
+```
+
+**Older docs**
+
+```shell
+# The new file.
+versioned_docs/version-1.0.0/new.md
+
+# Edit the corresponding sidebar file.
+versioned_sidebars/version-1.0.0-sidebars.json
+```
+
+### Linking files
+
+- Remember to include the `.md` extension.
+- Files will be linked to correct corresponding version.
+- Relative paths work as well.
+
+```md
+The [@hello](hello.md#paginate) document is great!
+
+See the [Tutorial](../getting-started/tutorial.md) for more info.
+```
+
+## Versions
+
+Each directory in `versioned_docs/` will represent a documentation version.
+
+### Updating an existing version
+
+You can update multiple docs versions at the same time. Because each directory in `versioned_docs/` represents specific routes when published.
+
+1. Edit any file.
+1. Commit and push changes.
+1. It will be published to the version.
+
+**Example:** When you change any file in `versioned_docs/version-2.6/`, it will only affect the docs for version `2.6`.
+
+### Deleting an existing version
+
+You can delete/remove versions as well.
+
+1. Remove the version from `versions.json`.
+
+Example:
+
+```diff {4}
+[
+  "2.0.0",
+  "1.9.0",
+- "1.8.0"
+]
+```
+
+2. Delete the versioned docs directory. Example: `versioned_docs/version-1.8.0`.
+3. Delete the versioned sidebars file. Example: `versioned_sidebars/version-1.8.0-sidebars.json`.
+
+## Recommended practices
+
+### Version your documentation only when needed
+
+For example, you are building a documentation for your npm package `foo` and you are currently in version 1.0.0. You then release a patch version for a minor bug fix and it's now 1.0.1.
+
+Should you cut a new documentation version 1.0.1? **You probably shouldn't**. 1.0.1 and 1.0.0 docs shouldn't differ according to semver because there are no new features!. Cutting a new version for it will only just create unnecessary duplicated files.
+
+### Keep the number of versions small
+
+As a good rule of thumb, try to keep the number of your versions below 10. **It is very likely** that you will have a lot of obsolete versioned documentation that nobody even reads anymore. For example, [Babel](https://babeljs.io/versions) for example is currently in version `7.x`, and it no longer maintains documentation for version `<= 6.x`. Keep it small 😊
```

Comment:
I don't think using Babel is a very good example. Their V6 site is built on another SSG (Jekyll?) and maybe that's why they're not linking it from their V7 built on Docusaurus. They do link to their 6.26.3 docs on https://babeljs.io/versions. Maybe we can mention https://getbootstrap.com/ or https://lodash.com/docs/4.17.15 as examples instead.

Reply:
Maybe using jest as reference is good